src/dash-adapter.js

```diff
@@ -365,10 +375,42 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
     }
     this._maybeSetFilters();
     this._maybeSetDrmConfig();
+    this._maybeBreakStalls();
     this._shaka.configure(this._config.shakaConfig);
     this._addBindings();
   }
 
+  _clearStallInterval(): void {
+    if (this._stallInterval) {
+      clearInterval(this._stallInterval);
+      this._stallInterval = null;
+    }
+  }
+
+  _stallSmartTVHandler(): void {
+    this._clearStallInterval();
+    let lastCurrentTime = this._videoElement.currentTime;
+    this._stallInterval = setInterval(() => {
+      if (lastCurrentTime === this._videoElement.currentTime) {
+        this._videoElement.currentTime = parseFloat(this._videoElement.currentTime.toFixed(1)) + 0.1;
+      } else {
+        this._clearStallInterval();
+      }
+    }, 2 * 1000);
+  }
+
+  /**
+   * register to event to break the stalls on smart TV
+   * @returns {void}
+   * @private
+   */
+  _maybeBreakStalls(): void {
+    if (this._config.playback.forceBreakStall) {
+      this._eventManager.listenOnce(EventType.STALLED, this._stallSmartTVHandler);
+      this._eventManager.listenOnce(EventType.SEEKED, this._stallSmartTVHandler);
```

Comment:
do we need/want to remove them after one handler is called?
If stalled occurred we can remove seeked and vide versa?

Reply:
I clear the stall interval if it's not stopped for each one of them.
Yes, the issue that we saw occur only on the beginning.